cli/file_fetcher.rs

```diff
@@ -579,6 +579,29 @@ fn map_js_like_extension(path: &Path, default: MediaType) -> MediaType {
       None => default,
       Some("jsx") => MediaType::JSX,
       Some("tsx") => MediaType::TSX,
+      // Because DTS files do not have a separate media type, or a unique
+      // extension, we have to "guess" at those things that we consider that
+      // look like TypeScript, and end with `.d.ts` are DTS files.
+      Some("ts") => {
+        if default == MediaType::TypeScript {
+          match path.file_stem() {
+            None => default,
+            Some(os_str) => {
+              if let Some(file_stem) = os_str.to_str() {
+                if file_stem.ends_with(".d") {
+                  MediaType::Dts
+                } else {
+                  default
+                }
+              } else {
+                default
+              }
+            }
+          }
+        } else {
+          default
+        }
+      }
```

Comment:
I see we don't have any unit tests in this module for `map_js_like_extension()` - which is unfortunate.  Would you mind adding a new unit test to nail down the logic being added here?

Reply:
Yeah, it would be nicer to have them seperate... but I think file_fetcher needs a good hose down anyways.